### DISCUSSION TITLE
How should I pass `lpszClassName` when creating a `WNDCLASS`?

### DISCUSSION THREAD

**Discussion by jgcodes2020** [microsoft/CsWin32#807] [Q&A]
Title explains all. I'm not sure what the lifetime of the associated string should be. If it must last until application termination, I'd have to make a special wrapper class to keep the name referenced until then. If not, I can just use a `fixed` block.

**Comment by AArnott** [comment]
Same lifetime that a native code equivalent would be, presumably. I don't know what that is. But if that's an extended duration, merely defining a wrapper class won't necessarily ensure the pointer remains valid, since the GC can move memory around. You'd need to use a `GCHandle` to effectively pin a memory location long-term.
